module/userProjects/helper.js

```diff
@@ -2280,7 +2339,558 @@ module.exports = class UserProjectsHelper {
+     * @name certificateReIssue 
+     * @param {String} projectId - projectId.
+     * @returns {JSON} certificate re-issued details.
+    */
+
+     static certificateReIssue(projectId) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                //  get project details project for which certificate re-issue required .
+                const userProject = await projectQueries.projectDocument({
+                    _id: projectId,
+                    status: CONSTANTS.common.SUBMITTED_STATUS,
+                    certificate: {$exists:true}  
+                });
+
+                //  if project details not found.
+                if (!userProject.length > 0) {
+                    throw {
+                        status: HTTP_STATUS_CODE['bad_request'].status,
+                        message: CONSTANTS.apiResponses.USER_PROJECT_NOT_FOUND
+                    };
+                }
+                
+                //  fetch user data using userId of project and calling the profile API
+                let userProfileData = await userProfileService.profileReadPrivate(userProject[0].userId);
+                if ( userProfileData.success && 
+                     userProfileData.data &&
+                     userProfileData.data.response &&
+                     userProfileData.data.response.userName &&
+                     userProfileData.data.response.userName !== ""
+                ) {
+                    userProject[0].userProfile.userName = userProfileData.data.response.userName;
+                } else {
+                    throw {
+                        status: HTTP_STATUS_CODE['bad_request'].status,
+                        message: CONSTANTS.apiResponses.USER_PROFILE_NOT_FOUND
+                    };
+                }
+                if ( userProject[0].certificate.transactionId ) {
+                    delete userProject[0].certificate.transactionId;
+                }
+                if ( userProject[0].certificate.osid ) {
+                    delete userProject[0].certificate.osid;
+                }
+                await kafkaProducersHelper.pushProjectToKafka(userProject[0]);
+                return resolve({ 
+                    success: true,
+                    message: CONSTANTS.apiResponses.PROJECT_CERTIFICATE_GENERATED,
+                    data : {
+                        _id :  userProject[0]._id
+                    }
+
+                });
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: {}
+                });
+            }
+        })
+    }
+
+
 };
+/**
+ * _subCriteriaValidation.
+ * @method
+ * @name _subCriteriaValidation 
+ * @param {Object} conditions - condition data.
+ * @param {String} expression - validation expression
+ * @returns {Boolean} validation.
+*/
+
+function _subCriteriaValidation(conditions, expression, data) {
```

Comment:
@VISHNUDAS-tunerlabs consider moving certificate validation entire logic to a new helper.